src/getTime.js

```diff
@@ -18,7 +18,9 @@
  * @returns {string} - valid time
  */
 function getTime(str) {
-  // write code here
+  const pattern = /(^|\D)((([0-1]\d)|(2[0-3])):[0-5]\d)(\D|$)/;
+
+  return pattern.test(str) ? str.match(pattern)[2] : '';
```

Comment:
why `str.match(pattern)[2]`?

Reply:
edit the pattern so you can use `[0]`index

Reply:
> edit the pattern so you can use `[0]`index

ok